pkg/streamingpromql/functions/common.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package functions
+
+import (
+	"github.com/grafana/mimir/pkg/streamingpromql/pooling"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type SeriesMetadataFunction func(seriesMetadata []types.SeriesMetadata, pool *pooling.LimitingPool) ([]types.SeriesMetadata, error)
+
+func DropSeriesName(seriesMetadata []types.SeriesMetadata, _ *pooling.LimitingPool) ([]types.SeriesMetadata, error) {
+	for i := range seriesMetadata {
+		seriesMetadata[i].Labels = seriesMetadata[i].Labels.DropMetricName()
+	}
+
+	return seriesMetadata, nil
+}
+
+type InstantVectorFunction func(seriesData types.InstantVectorSeriesData, pool *pooling.LimitingPool) (types.InstantVectorSeriesData, error)
+
+func FloatTransformationFunc(transform func(f float64) float64) InstantVectorFunction {
```

Comment:
Given this is only used by `FloatTransformationDropHistogramsFunc`, I'd just move it in there - if we need this separately in the future we can extract or duplicate it.

Reply:
I'd like to keep it separate since we unit test that it preserves the histograms. I will not export the function though.